### PR TITLE
Add spack repo feature to the helper script

### PR DESCRIPTION
**Pull request by rjanalik**

### CODE REVIEW

**Comment by rjanalik** [comment]
What about renaming `--with-repo` to `--add-repo`?

File: bin/spack-install-helper-10-preprocessing [hunk 1]
```diff
@@ -8,6 +8,22 @@
 # A100 = 8.0
 # H100 = 9.0
 
+SPACK_REPO_LIST=()
+
+while true
+do
+    case $1 in
+        --add-repo )
+            SPACK_REPO_LIST+=($2)
```
Comment (finkandreas) [inline bin/spack-install-helper-10-preprocessing]: probably never happening, but if someone ever comes up with weir whitespace directories, we would nee `"$2"`, i.e. quote the argument

File: bin/spack-install-helper-20-configure-environment [hunk 2]
```diff
@@ -24,3 +24,8 @@ if [[ -n ${NUM_PROCS} ]] ; then
 else
     echo "WARNING: NUM_PROCS is an empty variable. This is not the usual case, and it should be set to the number of processes that you want to use for building the spack software stack."
 fi
+
+for repo in ${SPACK_REPO_LIST[@]}
+do
+    spack repo add ${repo}
```
Comment (finkandreas) [inline bin/spack-install-helper-20-configure-environment]: same as above, it needs quotes, i.e. `"${repo}"`

**Review by finkandreas — CHANGES_REQUESTED** [review]
in general it is fine. Not sure if we want to support both syntaxes, `--add-repo=my_repo` and `--add-repo my_repo`. The former right now would not work. I guess you could make it work like
```bash
case $1 in
  --add-repo* )
    ...
    shift
    if [[ $1 == "--add-repo" ]] ; then
      shift
    fi
esac
```

However I am also fine with enforcing supporting only the latter syntax and disallowing the form `--some-arg=some-value` (this is more of a decision we need to make and stick to that decision)

File: bin/spack-install-helper-20-configure-environment [hunk 3]
```diff
@@ -24,3 +24,8 @@ if [[ -n ${NUM_PROCS} ]] ; then
 else
     echo "WARNING: NUM_PROCS is an empty variable. This is not the usual case, and it should be set to the number of processes that you want to use for building the spack software stack."
 fi
+
+for repo in ${SPACK_REPO_LIST[@]}
```
Comment (finkandreas) [inline bin/spack-install-helper-20-configure-environment]: Also here we need quotes, i.e. `"${SPACK_REPO_LIST[@]}`

**Review by finkandreas — CHANGES_REQUESTED** [review]

**Review by finkandreas — APPROVED** [review]
lgtm

**Pull request merged**